### PR TITLE
Add functional tests

### PR DESCRIPTION
**Pull request by SchSeba**
This commit add functional tests for the ptp feature.

The test where taken from https://github.com/openshift-kni/cnf-features-deploy/

Signed-off-by: Sebastian Sch <sebassch@gmail.com>

### CODE REVIEW

**Comment by SchSeba** [comment]
output

```
=== RUN   TestTest
Running Suite: PTP e2e integration tests
========================================
Random Seed: 1585663883
Will run 8 of 8 specs

•••
------------------------------
• [SLOW TEST:91.772 seconds]
ptp
/home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp.go:41
  PTP e2e tests
  /home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp.go:94
    PTP Interfaces discovery
    /home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp.go:97
      The interfaces support ptp can be discovered correctly
      /home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp.go:113
------------------------------
• [SLOW TEST:99.518 seconds]
ptp
/home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp.go:41
  PTP e2e tests
  /home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp.go:94
    PTP Interfaces discovery
    /home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp.go:97
      The virtual interfaces should be not discovered by ptp
      /home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp.go:126
------------------------------
• [SLOW TEST:32.785 seconds]
ptp
/home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp.go:41
  PTP e2e tests
  /home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp.go:94
    PTP Interfaces discovery
    /home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp.go:97
      PTP daemon apply match rule based on nodeLabel
      /home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp.go:138
------------------------------
• [SLOW TEST:6.509 seconds]
ptp
/home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp.go:41
  PTP e2e tests
  /home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp.go:94
    PTP Interfaces discovery
    /home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp.go:97
      Slave can sync to master
      /home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp.go:156
------------------------------
• [SLOW TEST:95.172 seconds]
ptp
/home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp.go:41
  Test NodeName Selector
  /home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp.go:189
    Check if Node has Profile
    /home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp.go:223
      PTP daemon can apply match rule based on nodeLabel
      /home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp.go:225
------------------------------

Ran 8 of 8 Specs in 337.685 seconds
SUCCESS! -- 8 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestTest (337.69s)
PASS
```

File: test/ptp.go [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+			ptpConfigTest := mutateProfile(ptpConfigSlave, ptpConfigName, nodes.Items[0].Name)
+
+			status := client.Client.CoreV1Interface.RESTClient().Post().AbsPath(configPtpAPIPath).
+				Resource("ptpconfigs").Body(ptpConfigTest).Context(context.TODO()).Do()
+			Expect(status.Error()).NotTo(HaveOccurred(), fmt.Sprint("PTP config creation Error"))
+
+			testPtpPod, err = getPtpPodOnNode(nodes.Items[0].Name)
+			Expect(err).NotTo(HaveOccurred())
+
+			testPtpPod, err = getPtpPodOnNode(nodes.Items[0].Name)
```
Comment (zshi-redhat) [inline test/ptp.go:216]: nit: This is executed twice, but it's good for testing :-)

File: test/ptp.go [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+					}
+				}
+				Expect(masterID).NotTo(BeNil())
+				Expect(slaveMasterID).NotTo(BeNil())
+				Expect(slaveMasterID).Should(HavePrefix(masterID), "Error match MasterID with the SlaveID. Slave connected to another Master")
+			})
+		})
+	})
+
+	Describe("Test NodeName Selector", func() {
```
Comment (zshi-redhat) [inline test/ptp.go:189]: It looks to me this is testing if high priority profile can be applied?

**Comment by fedepaol** [comment]
> so there are assumptions that before executing the tests, `slave` and `master` `PtpConfigs` shall be created and nodes are labeled with slave and grandmaster. Do you think we could doc it somwhere like in run-functests.sh or readme?

This PR is the first step (moving the ptp tests here). Once this gets in, we are going to modify the tests to both work for looking for existing configuration or setting the required configuration under a parameter.

File: test/ptp.go (hunk 1 above)
Comment (SchSeba) [reply test/ptp.go]: you are right thanks!

File: test/ptp.go (hunk 2 above)
Comment (SchSeba) [reply test/ptp.go]: I change the name in this test we test both the NodeName selector and the priority function

**Comment by SchSeba** [comment]
@zshi-redhat I am working on a second PR that will create the policies as part of the tests no need to document anything the tests should depend only on the operator to be installed.

**Comment by zshi-redhat** [comment]
> > so there are assumptions that before executing the tests, `slave` and `master` `PtpConfigs` shall be created and nodes are labeled with slave and grandmaster. Do you think we could doc it somwhere like in run-functests.sh or readme?
> 
> This PR is the first step (moving the ptp tests here). Once this gets in, we are going to modify the tests to both work for looking for existing configuration or setting the required configuration under a parameter.

sounds good, thanks for the comment!

**Comment by zshi-redhat** [comment]
> @zshi-redhat I am working on a second PR that will create the policies as part of the tests no need to document anything the tests should depend only on the operator to be installed.

Sure, let's merge this one, it looks good and test passed.

**Pull request merged**